### PR TITLE
prevent tests hangs in credentials-helper

### PR DESCRIPTION
**Pull request by campersau**
> The credentials-helper tests sometimes don't pass when I am having a crashed ungit instance opened in chrome because I stopped the server and the crashed ungit pages tries to reconnect by sending polling requests with socket.io which then the node server in the credentials-helper test picks up and throws inside of the request handler which means that the server does not a response and eventually times out. So to prevent this edge case we could add a timeout to the node response so it times out faster.

See https://github.com/FredrikNoren/ungit/pull/1365#issuecomment-629758116y

### CODE REVIEW

File: test/spec.credentials-helper.js [hunk 1]
```diff
@@ -11,26 +11,33 @@ describe('credentials-helper', () => {
...
+        const params = querystring.parse(reqUrl.query);
+        expect(params['remote']).to.be(`${remote}`);
+        expect(params['socketId']).to.be(`${socketId}`);
 
-      res.writeHead(200, { 'Content-Type': 'application/json' });
-      res.end(JSON.stringify(payload));
+        res.writeHead(200, { 'Content-Type': 'application/json' });
+        res.end(JSON.stringify(payload));
+      } finally {
+        if (!res.finished) {
```
Comment (Hirse) [inline test/spec.credentials-helper.js:25]: It seems that `finished` is deprecated: https://nodejs.org/api/http.html#http_response_finished
Maybe this should use `writableEnded` or `writableFinished` instead?

Comment (campersau) [reply test/spec.credentials-helper.js:25]: These two APIs were added in `v12.9.0` and `v12.7.0` so we have to use `finished` for now...

Comment (Hirse) [reply test/spec.credentials-helper.js:25]: Oh, I didn't see that part, I was too distracted by the big, red "Deprecated" banner. 😄
